src/shared/util.js

```diff
@@ -132,9 +132,9 @@ export function hasOwn (obj: Object | Array<*>, key: string): boolean {
  * Create a cached version of a pure function.
  */
 export function cached<F: Function> (fn: F): F {
-  const cache = Object.create(null)
+  const cache: Object = Object.create(null)
   return (function cachedFn (str: string) {
-    const hit = cache[str]
+    const hit: any = cache[str]
```

Comment:
Does this affect anything?

Reply:
I don't think there is any problem. 🤔 
In fact it can return various types of values.

case1: Regex
https://github.com/vuejs/vue/blob/e259fc306eece6be8014d2ace0258d9775971cd5/src/compiler/parser/text-parser.js#L9

case2: Object
https://github.com/vuejs/vue/blob/30258a9a60b96c25c95685cb17ee35838830ee4c/src/platforms/web/util/style.js#L5